tools/readme_config_includer/generator.go

```diff
@@ -0,0 +1,173 @@
+// This is a tool to embedd configuration files into the README.md of all plugins
+// It searches for TOML sections in the plugins' README.md and detects includes specified in the form
+//     ```toml [@includeA.conf[ @includeB[ @...]]
+//         Whatever is in here gets replaced.
+//     ```
+// Then it will replace everything in this section by the concatenation of the file `includeA.conf`, `includeB` etc.
+// content. The tool is not stateful, so it can be run multiple time with a stable result as long
+// as the included files do not change.
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log" //nolint:revive
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/text"
+)
+
+type includeBlock struct {
+	Includes []string
+	Start    int
+	Stop     int
+}
+
+func (b *includeBlock) extractBlockBorders(node *ast.FencedCodeBlock) {
+	// The node info starts at the language tag and stops right behind it
+	b.Start = node.Info.Segment.Stop + 1
+	b.Stop = b.Start
+
+	// To determine the end of the block, we need to iterate to the last line
+	// and take its stop-offset as the end of the block.
+	lines := node.Lines()
+	for i := 0; i < lines.Len(); i++ {
+		b.Stop = lines.At(i).Stop
+	}
+}
+
+func insertInclude(buf *bytes.Buffer, include string) error {
+	file, err := os.Open(include)
+	if err != nil {
+		return fmt.Errorf("opening include %q failed: %v", include, err)
+	}
+	defer file.Close()
+
+	// Write the include and make sure we get a newline
+	if _, err := io.Copy(buf, file); err != nil {
+		return fmt.Errorf("inserting include %q failed: %v", include, err)
+	}
+	return nil
+}
+
+func insertIncludes(buf *bytes.Buffer, b includeBlock) error {
+	// Insert all includes in the order they occured
+	for _, include := range b.Includes {
+		if err := insertInclude(buf, include); err != nil {
+			return err
+		}
+	}
+	// Make sure we add a trailing newline
+	if !bytes.HasSuffix(buf.Bytes(), []byte("\n")) {
+		if _, err := buf.Write([]byte("\n")); err != nil {
+			return errors.New("adding newline failed")
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	goPackage := os.Getenv("GOPACKAGE")
+	_ = goPackage
```

Comment:
Why do we need to do this? I don't see `goPackage` being used anywhere.

Reply:
Good catch! Seems to be a left-over... _Note to myself: DON'T DO THIS!_

Reply:
Fixed.